kafka/consumer.go

```diff
@@ -545,6 +553,9 @@ func (c *Consumer) Close() (err error) {
 		c.Poll(100)
 	}
 
+	// After this point, no more consumer methods may be called.
+	atomic.StoreUint32(&c.isClosed, 1)
```

Comment:
Should this be just before returning? `rd_kafka_destroy` can stuck due to various reasons.

Reply:
We're destroying an important queue below, which is necessary for most public API to run (it's the queue on which we're getting all the events from, from librdkafka side).

So most functions called after this point will panic due to segfault

Reply:
I was just talking about `atomic.StoreUint32(&c.isClosed, 1)`